game/src/main/kotlin/world/gregs/voidps/world/activity/dnd/shootingstar/ShootingStar.kts

```diff
@@ -0,0 +1,193 @@
+
+import world.gregs.voidps.engine.client.message
+import world.gregs.voidps.engine.client.variable.start
+import world.gregs.voidps.engine.data.definition.data.Rock
+import world.gregs.voidps.engine.entity.World
+import world.gregs.voidps.engine.entity.character.forceWalk
+import world.gregs.voidps.engine.entity.character.mode.interact.Interact
+import world.gregs.voidps.engine.entity.character.move.walkTo
+import world.gregs.voidps.engine.entity.character.npc.NPC
+import world.gregs.voidps.engine.entity.character.npc.NPCs
+import world.gregs.voidps.engine.entity.character.npc.npcOperate
+import world.gregs.voidps.engine.entity.character.player.Players
+import world.gregs.voidps.engine.entity.character.player.chat.ChatType
+import world.gregs.voidps.engine.entity.character.player.skill.Skill
+import world.gregs.voidps.engine.entity.character.setAnimation
+import world.gregs.voidps.engine.entity.obj.*
+import world.gregs.voidps.engine.entity.objectDespawn
+import world.gregs.voidps.engine.entity.worldSpawn
+import world.gregs.voidps.engine.inject
+import world.gregs.voidps.engine.inv.add
+import world.gregs.voidps.engine.inv.inventory
+import world.gregs.voidps.engine.inv.remove
+import world.gregs.voidps.engine.map.collision.blocked
+import world.gregs.voidps.engine.queue.softQueue
+import world.gregs.voidps.engine.timer.timerStart
+import world.gregs.voidps.engine.timer.toTicks
+import world.gregs.voidps.type.Direction
+import world.gregs.voidps.type.Tile
+import world.gregs.voidps.type.random
+import world.gregs.voidps.world.activity.dnd.shootingstar.ShootingStarHandler.currentActiveObject
+import world.gregs.voidps.world.activity.dnd.shootingstar.ShootingStarHandler.currentStarTile
+import world.gregs.voidps.world.activity.dnd.shootingstar.ShootingStarHandler.startEvent
+import world.gregs.voidps.world.activity.dnd.shootingstar.ShootingStarHandler.totalCollected
+import world.gregs.voidps.world.interact.dialogue.Cheerful
+import world.gregs.voidps.world.interact.dialogue.Sad
+import world.gregs.voidps.world.interact.dialogue.type.npc
+import world.gregs.voidps.world.interact.entity.combat.hit.damage
+import java.util.concurrent.TimeUnit
+import kotlin.math.roundToInt
+
+val objects: GameObjects by inject()
+val npcs: NPCs by inject()
+val players: Players by inject()
+
+worldSpawn {
+    eventUpdate()
+}
+
+fun eventUpdate() {
+    World.queue("shooting_star_event_timer", startEvent) {
+        if(isPlayersPresent()){
+            eventUpdate() // theres players present so skip this event removal
+        }
+       if (currentStarTile != Tile.EMPTY) {
+            cleanseEvent(true)
+            println("There was already an active star, deleted it and started a new event")
+       }
+       startCrashedStarEvent()
+       eventUpdate()
+    }
+}
+
+fun isPlayersPresent(): Boolean {
+    return currentStarTile.toCuboid(5, 5).any { tile ->
+        players[tile].isNotEmpty()
+    }
+}
```

Comment:
```suggestion
       if (currentStarTile != Tile.EMPTY) {
            cleanseEvent(true)
            println("There was already an active star, deleted it and started a new event")
       }
       startCrashedStarEvent()
       eventUpdate()
    }
}
```

Don't think we should do this the player has 10 minutes after, that should be enough time, (also could use zone rather than lookup every tile)